src/components/AxisCollection/index.js

```diff
@@ -1,72 +1,130 @@
-  static propTypes = {
-    width: PropTypes.number.isRequired,
-    height: PropTypes.number.isRequired,
-    series: seriesPropType,
-    zoomable: PropTypes.bool,
-    updateYTransformation: PropTypes.func,
-    yAxisWidth: PropTypes.number,
-    yTransformations: PropTypes.objectOf(
-      PropTypes.shape({
-        k: PropTypes.number.isRequired,
-        y: PropTypes.number.isRequired,
-        rescaleY: PropTypes.func.isRequired,
-      })
-    ).isRequired,
-  };
+const propTypes = {
+  height: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired,
+  series: seriesPropType,
+  zoomable: PropTypes.bool,
+  updateYTransformation: PropTypes.func,
+  yAxisWidth: PropTypes.number,
+  yTransformations: PropTypes.objectOf(
+    PropTypes.shape({
+      k: PropTypes.number.isRequired,
+      y: PropTypes.number.isRequired,
+      rescaleY: PropTypes.func.isRequired,
+    })
+  ).isRequired,
+  axisDisplayMode: axisDisplayModeType,
+  onMouseEnter: PropTypes.func,
+  onMouseLeave: PropTypes.func,
+};
 
-  static defaultProps = {
-    series: [],
-    zoomable: true,
-    updateYTransformation: () => {},
-    yAxisWidth: 50,
-  };
+const defaultProps = {
+  series: [],
+  zoomable: true,
+  updateYTransformation: () => {},
+  yAxisWidth: 50,
+  axisDisplayMode: AxisDisplayMode.ALL,
+  onMouseEnter: null,
+  onMouseLeave: null,
+};
 
-  shouldComponentUpdate() {
-    // TODO: Implement
-    return true;
-  }
+class AxisCollection extends React.Component {
+  showCollapsedAxis = () => {
+    const { series, axisDisplayMode } = this.props;
+    return series.filter(
+      s =>
+        !s.hidden &&
+        (s.yAxisDisplayMode || axisDisplayMode) === AxisDisplayMode.COLLAPSED
+    ).length;
+  };
 
-  render() {
+  renderAllAxes() {
     const {
-      width,
-      height,
+      axisDisplayMode,
       series,
       zoomable,
-      yAxisWidth,
+      height,
       updateYTransformation,
+      yAxisWidth,
       yTransformations,
     } = this.props;
-    let axisOffsetX = 0;
+    // Make sure that the axes don't appear if we're also going to be rendering
+    // the placeholder axis.
+    let axisOffsetX = this.showCollapsedAxis() ? yAxisWidth : 0;
+    return series
+      .filter(
+        s =>
+          !s.hidden &&
+          (s.yAxisDisplayMode || axisDisplayMode) === AxisDisplayMode.ALL
+      )
+      .map((s, idx) => {
+        if (idx > 0) {
+          axisOffsetX += yAxisWidth;
+        }
+        return (
+          <YAxis
+            key={`y-axis--${s.id}`}
+            offsetx={axisOffsetX}
+            zoomable={s.zoomable !== undefined ? s.zoomable : zoomable}
+            series={s}
+            height={height}
+            width={yAxisWidth}
+            updateYTransformation={updateYTransformation}
+            yTransformation={yTransformations[s.id]}
+          />
+        );
+      });
+  }
+
+  renderPlaceholderAxis() {
+    const { axisDisplayMode, height, yAxisWidth, series } = this.props;
```

Comment:
eslint screaming on axisDisplayMode and series unused variables, we should probably add pre commit hooks in a separate issue.

Reply:
Done